python/tvm/relay/op/annotation/annotation.py

```diff
@@ -22,7 +22,17 @@
 from .. import op as reg
 
 
-def on_device(data, device):
+def _device_to_int(device):
+    if isinstance(device, _Device):
+        return device.device_type
+    if isinstance(device, str):
+        return _nd.device(device).device_type
+    raise ValueError(
```

Comment:
I think this is just refactored from the below, but do we have a test for `ValueError` being raised?

Reply:
Can't find any existing unit tests for these so added some.